### ISSUE TITLE
MSAL SDK 1.2.9 crashing on handleMSALResponse

### ISSUE THREAD

**Issue by asmaan** [AzureAD/microsoft-authentication-library-for-objc#1659]
MSAL Logs

```
2023-02-24 14:46:36.263629 #NORMAL #ERROR : (TestClass.m:282) MSAL log: TID=446583 MSAL 1.2.9 iOS 16.1.1 [2023-02-24 14:46:36] Complete cert auth challenge with end URL: msauth.com.TestApp.TestApp://auth/broker?msg_protocol_ver=(not-null)&response=(not-null)&hash=(not-null)
2023-02-24 14:46:36.264623 #NORMAL #ERROR : (TestClass.m:282) MSAL log: TID=446583 MSAL 1.2.9 iOS 16.1.1 [2023-02-24 14:46:36] Handling broker response.
2023-02-24 14:46:36.267480 #NORMAL #ERROR : (TestClass.m:282) MSAL log: TID=446583 MSAL 1.2.9 iOS 16.1.1 [2023-02-24 14:46:36] Init MSIDKeychainTokenCache with keychainGroup: PCB253GE35.com.microsoft.adalcache
2023-02-24 14:46:36.267689 #NORMAL #ERROR : (TestClass.m:282) MSAL log: TID=446583 MSAL 1.2.9 iOS 16.1.1 [2023-02-24 14:46:36] Init MSIDKeychainTokenCache with keychainGroup: PCB253GE35.com.microsoft.adalcache
2023-02-24 20:16:40.945775+0530 TestApp[3396:446583] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'data parameter is nil'
```

**Comment by asmaan** [comment]
Crash happening at below line when additional info is nil

`            NSData *jsonData = [json[MSID_ADDITIONAL_EXTENSION_DATA_KEY] dataUsingEncoding:NSUTF8StringEncoding];
`

**Comment by swasti29** [comment]
We will fix this in next release of MSAL 1.2.10

**Comment by swasti29** [comment]
Hi @asmaan, 
We could not able to reproduce the issue at our end. Kindly help us with the full stacktrace.

Also, if you have UBSan (UndefinedBehaviorSanitizer) or related tools to check unexpected nil and crash the app. 
Does it happen everytime? Does it happen on both simulator and real device?

**Comment by swasti29** [comment]
The nil check has been available in MSAL release: 1.2.10.  Closing the issue for now. Feel free to reopen it. 
Also share the full stracktrace with us in case it reoccurs. Also confirm weather it happened everytime? Does it happen on both simulator and real device?
